pkg/rtc/mediatrackreceiver.go

```diff
@@ -97,32 +94,26 @@ type MediaTrackReceiverParams struct {
 }
 
 type MediaTrackReceiver struct {
-	params      MediaTrackReceiverParams
-	muted       atomic.Bool
-	simulcasted atomic.Bool
+	params MediaTrackReceiverParams
 
 	lock            sync.RWMutex
 	receivers       []*simulcastReceiver
-	receiversShadow []*simulcastReceiver
 	trackInfo       *livekit.TrackInfo
-	layerDimensions map[livekit.VideoQuality]*livekit.VideoLayer
 	potentialCodecs []webrtc.RTPCodecParameters
 	state           mediaTrackReceiverState
 
 	onSetupReceiver     func(mime string)
 	onMediaLossFeedback func(dt *sfu.DownTrack, report *rtcp.ReceiverReport)
-	onVideoLayerUpdate  func(layers []*livekit.VideoLayer)
```

Comment:
Checked with Shishir on this and it is not useful. `TrackPublished` event is useful to know the maximum layer changing (start/stop due to congestion or dynacast). This was reporting all layers and not useful. Hence removing.